extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/theta/SketchBufferAggregator.java

```diff
@@ -30,17 +30,16 @@
 
 import java.nio.ByteBuffer;
 import java.util.HashMap;
+import java.util.IdentityHashMap;
 import java.util.Map;
 
 public class SketchBufferAggregator implements BufferAggregator
 {
   private final ObjectColumnSelector selector;
   private final int size;
   private final int maxIntermediateSize;
-
-  private NativeMemory nm;
-
   private final Map<Integer, Union> unions = new HashMap<>(); //position in BB -> Union Object
```

Comment:
@akashdw this is a simple change and would actually improve performance in this case, so I would +1 this one